docs/csharp/tutorials/generate-consume-asynchronous-stream.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Generate and consume async streams
+description: This advanced tutorial illustrates scenarios where generating and consuming async streams provides a more natural way to work with sequences of data that may be generated asynchronously.
+ms.date: 02/10/2019
+ms.custom: mvc
+---
+# Tutorial: Generate and consume async streams using C# 8.0 and .NET Core 3.0
+
+C# 8.0 introduces **async streams**, which model a streaming source of data when the elements in the data stream may be retrieved or generated asynchronously. Async streams rely on new interfaces introduced in .NET Standard 2.1 and implemented in .NET Core 3.0 to provide a natural programming model for asynchronous streaming data sources.
+
+In this tutorial, you'll learn how to:
+
+> [!div class="checklist"]
+> * Create a data source that generates a sequence of data elements asynchronously.
+> * Consume that data source asynchronously.
+> * Recognize when the new interface and data source are preferred to earlier synchronous data sequences.
+
+## Prerequisites
+
+You’ll need to set up your machine to run .NET Core, including the C# 8.0 beta compiler. The C# 8 beta compiler is available with [Visual Studio 2019 preview 1](https://visualstudio.microsoft.com/vs/preview/), or [.NET Core 3.0 preview 1](https://dotnet.microsoft.com/download/dotnet-core/3.0). Async streams are first available in .NET C.0 preview 2.
```

Comment:
Suggestion to say starting with otherwise it seems it's only available with that. Also fixed the version number.

Also somewhere in the prereqs you should mention they need the SDK, so they know what to download.

```suggestion
You’ll need to set up your machine to run .NET Core, including the C# 8.0 beta compiler. The C# 8 beta compiler is available starting with [Visual Studio 2019 preview 1](https://visualstudio.microsoft.com/vs/preview/), or [.NET Core 3.0 preview 1](https://dotnet.microsoft.com/download/dotnet-core/3.0). Async streams are first available in .NET 3.0 preview 2.
```